snapshot.go

```diff
@@ -249,3 +252,35 @@ func (r *Raft) compactLogs(snapIdx uint64) error {
 	}
 	return nil
 }
+
+// resetMonotonicLogs truncates all logs for MonotonicLogStores after a snapshot
+// restore.
+func (r *Raft) resetMonotonicLogs() error {
+	// Only truncate if the underlying store has monotonic log index
+	// requirements.
+	logs, ok := r.logs.(MonotonicLogStore)
+	if ok && logs.IsMonotonic() {
+		defer metrics.MeasureSince([]string{"raft", "resetMonotonicLogs"}, time.Now())
```

Comment:
nit: maybe we shouldn't leak this complicated internal name to operators to whom it might not make sense?

How about calling this "resetLogs". In Consul we'd document it as only applying to certain log stores or something.